src/hb-common.cc

```diff
@@ -665,6 +666,125 @@ hb_script_get_horizontal_direction (hb_script_t script)
   return HB_DIRECTION_LTR;
 }
 
+static int
+lang_compare_first_component (const void *pa,
+                              const void *pb)
+{
+  const char *a = (const char *) pa;
+  const char *b = (const char *) pb;
+  unsigned int da, db;
+  const char *p;
+
+  p = strstr (a, "-");
+  da = p ? (unsigned int) (p - a) : strlen (a);
+
+  p = strstr (b, "-");
+  db = p ? (unsigned int) (p - b) : strlen (b);
+
+  return strncmp (a, b, hb_max (da, db));
+}
+
+static const HbScriptForLang *
+find_best_lang_match (hb_language_t language)
+{
+  const char *lang_str;
+  const char *record, *start, *end;
+  const char *records;
+  unsigned int record_size;
+  unsigned int num_records;
+
+  if (language == nullptr)
+    return nullptr;
```

Comment:
guess this can be moved to top so assignments can be merged with declarations as we are in C++?